expat/doc/reference.html

```diff
@@ -1112,12 +1113,40 @@ <h4 id="XML_Parse">XML_Parse</h4>
 <code>isFinal</code> parameter informs the parser that this is the last
 piece of the document. Frequently, the last piece is empty (i.e.
 <code>len</code> is zero.)
+</p>
+
+<p>
 If a parse error occurred, it returns <code>XML_STATUS_ERROR</code>.
 Otherwise it returns <code>XML_STATUS_OK</code> value.
 Note that regardless of the return value, there is no guarantee that all
 provided input has been parsed; only after <a href="#isFinal">the
 concluding call</a> will all handler callbacks and parsing errors have
 happened.
+</p>
+
+<p>
+Function <code>XML_Parse</code> is mostly a convenience wrapper pairing
+calls to <code><a href="#XML_ParseBuffer">XML_ParseBuffer</a></code>
+and <code><a href="#XML_GetBuffer">XML_GetBuffer</a></code>.
+You can roughly imagine <code>XML_Parse</code>'s implementation as
+a more complete version with error checking of the following:
+</p>
+
+<pre>
+enum XML_Status XMLCALL
+XML_Parse(XML_Parser parser, const char *s, int len, int isFinal) {
+  // Not the actual XML_Parse, oversimplified!
+  void *buff = XML_GetBuffer(parser, len);
+  memcpy(buff, s, len);
+  return XML_ParseBuffer(parser, len, isFinal);
+}</pre>
```

Comment:
Do we need this much detail? Maybe the code can be removed if we just mention the copy in the paragraph above, e.g.

> XML_Parse is mostly a convenience wrapper that calls XML_GetBuffer, copies input from `s`, and parses it with XML_ParseBuffer.

Reply:
Personally I feel the code could not say it any clearer, a bit like a picture would say more than words, at least that was the idea. Maybe it's a question of view :shrug:

Reply:
The code is clear, but incomplete, and despite the comment warning about it, may be misleading: 
* it seemingly precludes the "XML_Parse may be faster" case
* someone reading uncarefully may take it as a "how to use Get/Parsebuffer"

Reply:
@Snild-Sony alright, I will trust your instinct on that code doing more harm then good and use a text version like yours instead.